src/realm/decimal128.cpp

```diff
@@ -1412,29 +1412,24 @@ Decimal128::Decimal128(double val, RoundTo rounding_precision)
     }
 }
 
-void Decimal128::from_int64_t(int64_t val)
+Decimal128::Decimal128(int val)
+    : Decimal128(static_cast<int64_t>(val))
+{
+}
+
+Decimal128::Decimal128(int64_t val)
 {
     constexpr uint64_t expon = uint64_t(DECIMAL_EXPONENT_BIAS_128) << 49;
     if (val < 0) {
         m_value.w[1] = expon | 0x8000000000000000ull;
-        m_value.w[0] = ~val + 1;
+        m_value.w[0] = val == std::numeric_limits<int64_t>::lowest() ? val : ~val + 1;
```

Comment:
`~std::numeric_limits<int64_t>::lowest() + 1` is a signed integer overflow, which is UB but in practice gives `std::numeric_limits<int64_t>::lowest()`.